### DISCUSSION TITLE
When will gpt-4 and GPT-4-32K be supported

### DISCUSSION THREAD

**Discussion by traitors09** [betalgo/openai#165] [General]
When will gpt-4 and GPT-4-32K be supported？

`    /// <summary>
    ///     More capable than any GPT-3.5 model, able to do more complex tasks, and optimized for chat. Will be updated with
    ///     our latest model iteration.
    ///     8,192 tokens	Up to Sep 2021
    /// </summary>
    public static string Gpt4 => "gpt-4";

    /// <summary>
    ///     Snapshot of gpt-4 from March 14th 2023. Unlike gpt-4, this model will not receive updates, and will only be
    ///     supported for a three month period ending on June 14th 2023.
    ///     8,192 tokens	Up to Sep 2021
    /// </summary>
    public static string Gpt_4_0314 => "gpt-4-0314";

    /// <summary>
    ///     Same capabilities as the base gpt-4 mode but with 4x the context length. Will be updated with our latest model
    ///     iteration.
    ///     32,768 tokens	Up to Sep 2021
    /// </summary>
    public static string Gpt_4_32k => "gpt-4-32k";

    /// <summary>
    ///     Snapshot of gpt-4-32 from March 14th 2023. Unlike gpt-4-32k, this model will not receive updates, and will only be
    ///     supported for a three month period ending on June 14th 2023.
    ///     32,768 tokens	Up to Sep 2021
    /// </summary>
    public static string Gpt_4_32k_0314 => "gpt-4-32k-0314";`

**Comment by kayhantolga** [comment]
What do you mean? The library supports new models, but you can't use them unless openAI gave you access. Did you join the waiting list?

**Comment by traitors09** [comment]
I saw that ChatGPT-4 support was not checked in the Features and assumed it was unfinished

**Reply by kayhantolga** [reply]
Oh, I see, it is because we do not have documentation for how we can pass an Image to chat GPT-4. You can use GPT-4 models just with text. at the moment
